lib/awful/tag.lua

```diff
@@ -465,6 +465,7 @@ function tag.object.set_screen(t, s)
 
     -- Change the screen
     tag.setproperty(t, "screen", s)
+    tag.setproperty(t, "index", #s:get_tags(true))
```

Comment:
While both methods causes race conditions, what you propose is worst as it will cause 2 tags to have to same index for a short time interval. Also, before you ask, setting the screen to `nil` to solve this also isn't a good idea have have its fair share of problems. There is already a possibility that this line does nothing at all because some other code will query the index and it will be re-generated elsewhere (see `get_index()`). However, this only happen if something explicitly query the index, so this line is required for the signal to be emitted.